### PR TITLE
Feature/config file

### LINKED ISSUE #43: Support .filenamelintrc file with one "format" option.
Other types of files to support:
- .filenamelintrc.json
- .filenamelintrc.js

### PR DESCRIPTION
**Pull request by bigsergey**
Fixes #43

### CODE REVIEW

File: src/options/get-options-from-file.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  const configFilePath = path.join(process.cwd(), configFileName);
+
+  if (fs.existsSync(configFilePath)) {
+    const content = fs.readFileSync(configFilePath, 'utf8');
+
+    if (isFileEmpty(content)) {
+      return {};
+    }
+
+    const options = JSON.parse(content);
```
Comment (erykpiast) [inline src/options/get-options-from-file.ts]: Maybe `try ... catch` and handle this error gracefully? I'd be better than `unexpected token at line...`

File: package.json [hunk 2]
```diff
@@ -14,22 +14,15 @@
...
   "lint-staged": {
-    "*.ts": [
-      "eslint --fix",
-      "git add"
-    ],
-    "*.md": [
-      "markdownlint --fix",
-      "git add"
-    ]
+    "*.ts": "eslint --fix",
```
[the remark this answers is not in the document]
Comment (bigsergey) [reply package.json]: yes, `git add` is default now

File: src/options/get-options-from-file.ts (hunk 1 above)
Comment (bigsergey) [reply src/options/get-options-from-file.ts]: nice catch, i will do it

File: src/options/__tests__/index.spec.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  describe('default options', () => {
+    beforeEach(async () => {
+      await mockedGetOptionsFromFile.mockResolvedValue({});
+    });
+
+    test('should return default options when custom options are empty', async () => {
+      expect(await getOptions()).toEqual(defaultOptions);
+    });
+
+    test('should return default ignore option when all options are undefined', async () => {
```
Comment (erykpiast) [inline src/options/__tests__/index.spec.ts]: what about `format`?

**Pull request merged**